docs/_posts/2021-08-31-luau-recap-august-2021.md

```diff
@@ -0,0 +1,112 @@
+
+[Cross-posted to the [Roblox Developer Forum](https://devforum.roblox.com/t/luau-recap-august-2021/).]
+
+## Editor features
+
+The Roblox Studio [Luau-Powered Autocomplete & Language Features Beta](https://devforum.roblox.com/t/script-editor-luau-powered-autocomplete-language-features-beta) that our team has been working on has finally been released!
+Be sure to check that out and leave your feedback for things we can improve.
+
+To support that feature, a lot of work went into:
+* Improving fault-tolerant parser recovery scenarios
+* Storing additional information in the AST, including comments, better location information and partial syntax data
+* Tracking additional information about types and their fields, including tracking definition locations, function argument names, deprecation state and custom Roblox-specific tags
+* Updating reflection information to provide more specific Instance types and correct previously missing/wrong type annotations
+* Hybrid typechecking mode to try and infer types even in scripts with no typechecking enabled
+* Support for types that are attached to the DataModel tree elements to provide instance member information
+* Placing limits to finish typechecking in a finite space/time.
+* Adding Autocomplete API for the Roblox Studio to get location-based entity information and appropriate suggestions
+* Additional type inference engine improvements and fixes
+
+While our work continues to respond to the feedback we receive, our team members are shifting focus to add generic functions, extend Parallel Luau, improve Lua VM performance, provide documentation and support typechecking of Lua metatable-based OOP constructs. 
+
+## Typechecking improvements
+
+Type constraint resolver now remembers constraints placed on individual table fields.
+
+This should fix false-positive errors reported after making sure the optional table field is present:
+```lua
+--!strict
+local t: {value: number?} = {value = 2}
+
+if t.value then
+    local v: number = t.value -- ok
+end
+```
+
+And it can also refine field type to a more specific one:
+```lua
+--!strict
+local t: {value: string|number} = {value = 2}
+
+if type(t.value) == "number" then
+    return t.value * 2 -- ok
+end
+```
+
+Like before, combining multiple conditions using 'and' and 'not' is also supported.
+
+---
+
+Constructing arrays with different values for optional/union types are now also supported for individual table fields and in functions call arguments:
+```lua
+--!strict
+type Foo = {x: number | string, b: number?}
+
+local function foo(l: {Foo}) end
+
+foo({
+	{x = 1234567},
+	{x = "hello"}, -- now ok
+})
+
+type Bar = {a: {Foo}}
+
+local foos: Bar = {a = {
+	{x = 1234567},
+	{x = "hello", b = 2}, -- now ok
+}}
+```
+
+---
+
+Finally, we have fixed an issue with Roblox class field access using indexing like `part["Anchored"] = true`.
+
+## Linter improvements
+
```

Comment:
Did we already announce the "duplicate condition" lint?

Reply:
Yes, that was last month